From: Sven Roederer <devel-sven@geroedel.de>
Date: Thu, 8 Apr 2021 21:19:46 +0200
Subject: imagebuilder: unset BINARY_FOLDER and DOWNLOAD_FOLDER in imagebuilder archive

Setting these folder during buildtime will embedd their location in the .config file.
This results in trying to use the same subfolders on the (other) machine running the
imagebuilder, where these might be not accessible at all.

mkdir: cannot create directory '/mnt/build': Permission denied
Makefile:116: recipe for target '_call_image' failed
make[2]: *** [_call_image] Error 1
Makefile:241: recipe for target 'image' failed
make[1]: *** [image] Error 2

Signed-off-by: Sven Roederer <devel-sven@geroedel.de>

diff --git a/target/imagebuilder/Makefile b/target/imagebuilder/Makefile
index f9c08776a8a68a3b724f32059851e307cdc2a37a..e79c0db46e186ba33674b72298751b23c1a0467a 100644
--- a/target/imagebuilder/Makefile
+++ b/target/imagebuilder/Makefile
@@ -29,6 +29,8 @@ $(BIN_DIR)/$(IB_NAME).tar.xz: clean
 	mkdir -p $(IB_KDIR) $(IB_LDIR) $(PKG_BUILD_DIR)/staging_dir/host/lib \
 		$(PKG_BUILD_DIR)/target $(PKG_BUILD_DIR)/scripts $(IB_DTSDIR)
 	-cp $(TOPDIR)/.config $(PKG_BUILD_DIR)/.config
+	$(SED) 's/^CONFIG_BINARY_FOLDER=.*/# CONFIG_BINARY_FOLDER was reset by Imagebuilder/' $(PKG_BUILD_DIR)/.config
+	$(SED) 's/^CONFIG_DOWNLOAD_FOLDER=.*/# CONFIG_DOWNLOAD_FOLDER was reset by Imagebuilder/' $(PKG_BUILD_DIR)/.config
 	$(CP) -L \
 		$(INCLUDE_DIR) $(SCRIPT_DIR) \
 		$(TOPDIR)/rules.mk \
